api/base/filters.py

```diff
@@ -134,7 +134,7 @@ def _validate_operator(self, field, field_name, op):
                     detail="Field '{0}' does not support match operators in a filter.".format(field_name)
                 )
 
-    def _parse_date_param(self, field, field_name, op, value):
+    def _parse_date_param(self, field, source_field_name, op, value):
```

Comment:
Can't you use convert_key() for this? https://github.com/centerforopenscience/osf.io/blob/f9e0eb63581d642956172dd3fb662e240ba334c0/api/base/filters.py#L215-L224